### PR TITLE
feat : login-signup-auth, add: private-routes

### PR DESCRIPTION
**Pull request by BelideNikhil**
# added authentication
+ login
+ signup
+ error validation 
+ hide login/signup routes when user is authenticated
+ hide cart/Wishlist routes when user is not authenticated

Preview: https://ninty-seven-x-store.netlify.app/login

### CODE REVIEW

File: src/Context/AuthContext.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    }
+}
+function AuthProvider({ children }) {
+    const [authState, authDispatchFuntion] = useReducer(authReducerFunction, initialAuthState);
+    const navigate = useNavigate();
+
+    const userLoginHandler = async (userData) => {
+        const { status, data } = await getLoginDetails(userData);
+        if (status === 200) {
+            lcStorgaeSetter(200, data);
```
Comment (msdeshmukh009) [inline src/Context/AuthContext.js]: Naming can be better.

**Review by msdeshmukh009 — APPROVED** [review]
The code is looking clean. The confirm password error is not showing up you might want to check that out.

File: src/App.js [hunk 2]
```diff
@@ -1,22 +1,46 @@
...
+                    </>
+                ) : (
+                    <>
+                        <Route path="/login" element={<Navigate to="/" />} />
+                        <Route path="/signup" element={<Navigate to="/" />} />
+                    </>
+                )}
+                <Route path="*" element={<Navigate replace to="/" />} />
+                <Route path="/mock" element={<Mockman />} />
+            </Routes>
```
Comment (Chakravarthi0) [inline src/App.js]: suggestion: you can extract the routes into a component

**Review by Chakravarthi0 — APPROVED** [review]
everthing looks good.

File: src/Context/AuthContext.js (hunk 1 above)
Comment (BelideNikhil) [reply src/Context/AuthContext.js]: Updated..

File: src/App.js (hunk 2 above)
Comment (BelideNikhil) [reply src/App.js]: Done!

**Pull request merged**